packages/kbn-expandable-flyout/src/components/render_menu.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiButtonGroup,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutProps,
+  EuiPanel,
+  EuiPopover,
+} from '@elastic/eui';
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  RENDER_MENU_BUTTON_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_OVERLAY_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_PUSH_TEST_ID,
+  RENDER_MENU_FLYOUT_TYPE_BUTTON_GROUP_TEST_ID,
+} from './test_ids';
+
+const RENDER_MENU_ICON_BUTTON = i18n.translate('expandableFlyout.renderMenu.popoverButton', {
+  defaultMessage: 'Open flyout render menu',
+});
+const FLYOUT_TYPE_TITLE = i18n.translate('expandableFlyout.renderMenu.flyoutTypeTitle', {
+  defaultMessage: 'Flyout type',
+});
+const FLYOUT_TYPE_OVERLAY_MODE = i18n.translate('expandableFlyout.renderMenu.overlayMode', {
+  defaultMessage: 'Overlay',
+});
+const FLYOUT_TYPE_PUSH_MODE = i18n.translate('expandableFlyout.renderMenu.pushMode', {
+  defaultMessage: 'Push',
+});
+const FLYOUT_TYPE_RESET_BUTTON = i18n.translate('expandableFlyout.renderMenu.resetButton', {
+  defaultMessage: 'Reset to default',
+});
+
+interface RenderMenuProps {
+  /**
+   * Current flyout type
+   */
+  flyoutTypeProps: {
+    /**
+     * 'push' or 'overlay'
+     */
+    type: EuiFlyoutProps['type'];
+    /**
+     * Callback to change the flyout type
+     */
+    onChange: (type: EuiFlyoutProps['type']) => void;
+    /**
+     * Disables the button group for flyout where the option shouldn't be available
+     */
+    disabled: boolean;
+  };
+}
+
+/**
+ * Renders a menu to allow the user to customize the flyout.
+ * Current customization are:
+ * - Flyout type: overlay or push
+ */
+export const RenderMenu: React.FC<RenderMenuProps> = ({ flyoutTypeProps }: RenderMenuProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const togglePopover = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const panels = [
```

Comment:
what do you think about moving the panels to closer to the alert flyout code? This looks too specific to be in the package for me.. i could imagine UI changes or more configs down the road, and we'd have to modify the package... like if we want to add a tooltip that says (push is disabled for timeline), we have to do it here?

Reply:
I went back and forth on this a few times... I thought about the tooltip you're mentioning and that almost tipped the balance towards having this code outside of the package, but ultimately, what I like about having it inside was:
- we control the UI (so we don't end up with n versions of a way to do push/overlay and all the other options we'll provide)
- this really seems to me like a package code, as it directly touches how the packages works internally. By that I mean that if for some reason we decide to switch `EuiFlyout` to something else, we don't want developers to have to know the name of the property to use, or even how it works... Granted that they kinda have to know today because they can pass the type (`'push'` or `'overlay'`), but it's typed and kept to a minimum...

My philosophy was let's keep it tightly coupled with the package for now, and when people complain about a limitation with a design that they have, we should be able to easily extract it into a reusable component, and make the popover in the package render whatever is passed in...

Happy to discuss this more though, great comment!

Reply:
sure, we can discuss more and brainstorm together. My main concern is the setting menu currently is not very flexible:
- ideally we should be able pass the `panels` and decide what settings to show, i.e. host or alert specific settings
- if other devs don't care about push vs overlay, the setting is still showing
- all the future config will sit in the package, it will be a pain to manage if we have to review settings outside of investigations

However, it is nice to have some package level control out of the box, so i understand why you take this approach. 

If no additional settings are planned for the next cycle or two, it's probably okay. just don't want to see the work here reverted in a couple of months :)

Reply:
I actually just tried the opposite way, where I move the entirety of the code within the package, and the live only the decision to enable/disable the functionality to the security solution (or other) plugin(s).

I like this approach, but I still see valid point of doing it like you suggested. Let's discuss this this week!

Reply:
Yea, I agree with Christine here. Consumers of the flyout package may want it to behave more like `pushBreakpoint={800} // innerWidth of window at which point the flyout enters "push" mode` rather than have a toggleable push vs overlay option. As an aside, I figure design at some point will want some kind of icon for defining push vs overlay if that's the plan as well. Something like one of these. When push is enabled, the icon is highlighted/depressed, when it's disabled it looks normal...

![image](https://github.com/user-attachments/assets/5020a942-1ff4-49ab-a9e6-86db6814ae77)

Reply:
I don't think the `pushBreakpoint={800}` approach you're suggestion would make sense for us right now, because at this time there is a single flyout for the entire Security Solution application. While it could make sense to have for the threshold to be at 800 on the alerts page, maybe on the rule details or the host or user pages it doesn't make any sense at all and we would want a completely different value...
I like the approach I'm taking in the second commit because now the functionality is fully internal to the package. Developers don't have to know about how it works, how to save it...

Also having the threshold customizable makes little sense to me if you combine the push/overlay functionality for a resizable flyout ([coming shortly](https://github.com/elastic/kibana/pull/182614) after this PR). If the flyout is resizable, I think it makes more sense to just let the user toggle on/off the push vs overlay. It would be much more complex (in code and in the UI) to allow to manage resizable as well as a `pushBreakPoint` value. Would we want those to be different?

I understand that the current approach is different from Discover which manages push/overlay by the window width, but they also have a single page their flyout is accessible from. We have the entire Security Solution application...
Happy to discuss this more though because I'm sure I'm not seeing the whole picture!

Reply:
Just documenting that we spoke offline about this. To re-iterate your arguments here make sense especially with the introduction of the resizable flyout. Providing users the flexibility to choose the option makes the most sense, and allowing developers to just enable/disable the functionality also makes sense. Looking forward, if we were to enable a "kiosk" mode then the "flyout type" will probably be  `overlay`, `push`, `modal / kiosk / something`. Can we rename "flyout type" to "layout type" or "layout setting" or just "layout"... When I hear flyout type, I think alert vs event vs entity, etc...

Reply:
I decided to keep flyout `type` for now for 2 reasons:
- while for us in SecuritySolution the word `type` could mean signal, alert, event... it wouldn't mean that for anyone else. This code leaves in the `packages` folder, so far enough I think from Security Solution
- more importantly, `type` is what EUI uses everywhere in the code and in their docs. It makes a lot of sense to use their terminology, because that's directly what we're using